autonomy/configurations/base.py

```diff
@@ -56,6 +56,73 @@
 }
 
 
+def is_strict_list(data: List) -> bool:
+    """
+    Check if a data list is an strict list
+
+    The data list contains a mapping object we need to process it as an
+    object containing configurable parameters. For example
+
+    cert_requests:
+      - public_key: example_public_key
+
+    This will get exported as `CONNECTION_NAME_CERT_REQUESTS_0_PUBLIC_KEY=example_public_key`
+
+    Where as
+
+    parameters:
+     - hello
+     - world
+
+     will get exported as `SKILL_NAME_PARAMETERS=["hello", "world"]`
+
+    :param data: Data list
+    :return: Boolean specifying whether it's a strict list or not
+    """
+    is_strict = True
+    for obj in data:
+        if isinstance(obj, dict):
+            return False
+        if isinstance(obj, list):
+            if not is_strict_list(data=obj):
+                return False
+    return is_strict
+
+
+def generate_env_vars_recursively(
```

Comment:
I've extracted this method to `open-aea` to make sure this method is consistent with `aea.helpers.env_vars.apply_env_variables` in this PR https://github.com/valory-xyz/open-aea/pull/460